src/librustdoc/html/render/print_item.rs

```diff
@@ -511,10 +511,16 @@ fn item_module(w: &mut Buffer, cx: &mut Context<'_>, item: &clean::Item, items:
                     _ => "",
                 };
 
-                let visibility_emoji = match myitem.visibility(tcx) {
+                let visibility_and_hidden = match myitem.visibility(tcx) {
                     Some(ty::Visibility::Restricted(_)) => {
-                        "<span title=\"Restricted Visibility\">&nbsp;🔒</span> "
+                        if myitem.is_doc_hidden() {
```

Comment:
Isn't the condition reversed here? Also, if the item is both hidden and private, should we show that it is private?

Reply:
oops. fixed. And yes, I think we should show both, though really hidden+private doesn't make sense.

Reply:
Then what's the use case of showing both?

Reply:
There's not a large use case, but it may make sense for crates that more regularly use `--document-private-items`.

Reply:
Well, we can always change it afterwards if we change our minds about it.

Reply:
(I meant it's fine to leave it as is)